### ISSUE TITLE
Firebase Gif URL with FLAnimatedImageView

### ISSUE THREAD

**Issue by Ankish** [rs/SDWebImage#2203]
### New Issue Checklist

* [x] I have read and understood the [CONTRIBUTING guide](https://github.com/rs/SDWebImage/blob/master/.github/CONTRIBUTING.md)
* [x] I have read the [Documentation](http://cocoadocs.org/docsets/SDWebImage/)
* [x] I have searched for a similar issue in the [project](https://github.com/rs/SDWebImage/issues) and found none

### Issue Info

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform Name           | e.g. ios / tvos
 Platform Version        | e.g. 8.0
 SDWebImage Version      | e.g. 3.7.6
 Integration Method      | e.g. carthage / cocoapods / manually
 Xcode Version           | e.g. Xcode 7.3
 Repro rate              | e.g. all the time (100%) / sometimes x% / only once
 Repro with our demo prj | e.g. does it happen with our demo project?
 Demo project link       | e.g. link to a demo project that highlights the issue

### Issue Description and Steps

Please fill in the detailed description of the issue (full output of any stack trace, compiler error, ...) and the steps to reproduce the issue.

**There is a provision for FLAnimatedImageView and Firebase. How do we load gifs at firebase url with  FLAnimatedImageView?**

**Comment by dreampiggy** [comment]
I'm not familiar with Firebase's cloud service. Does that means you should pass addtional HTTP Headers or something like HTTP Authorization ? See [headersFilter](https://github.com/rs/SDWebImage/blob/master/SDWebImage/SDWebImageDownloader.h#L162-L168) or [HTTP Authorization](https://github.com/rs/SDWebImage/blob/master/SDWebImage/SDWebImageDownloader.h#L152-L160)

Or you can provide the documentation for Firebase service to let I know the concept.

**Comment by Ankish** [comment]
Firebase has a support with SDWebImage. But FLAnimatedImageView and firebase is not linked.
Check doc at : https://firebase.google.com/docs/storage/ios/download-files

**Comment by dreampiggy** [comment]
Well. I know. It seems that FirebaseUI framework create their category for SDWebImage. It works for normal `UIImageView` but not for `FLAnimatedImageView`. This may not works because they use `sd_imageWithData`, which just present a `UIImage` but not `FLAnimatedImage`.

If you still want to use GIF with `FLAnimateImageView` and `FirebaseUI`. You should check [their issue](https://github.com/firebase/FirebaseUI-iOS/issues/278) which contains a custom category to suppor this.

If you do not want to use GIF with `FLAnimateImageView` but for all `UIImageView` like 3.7.6 version, you should add the GIF coder, see [GIF coder](https://github.com/rs/SDWebImage/wiki/Advanced-Usage#gif-coder)
